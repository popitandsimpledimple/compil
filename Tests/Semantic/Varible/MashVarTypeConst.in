program main;
var
    e,w: double;
    x,u: integer;
    y,m: string;
const 
    a = 5;
    b:string = 'hello world';
type 
    int = integer;
    MyString = string;

var pp: array[5..20] of Integer;
    oo: array[7..30] of array[0..40] of integer;
    ii: array[5..50, 0..60, 0..70] of integer;
    var rec: record aaa: Integer; dddd: Double; bbbb: Boolean; ssss: String; rrrr: record field: Integer; end; end;

begin 
write(pp[4], oo[10][10], ii[10+10][20*2][12+3*2]);

write(
rec.dddd, 
rec.bbbb, 
rec.ssss);

read(
rec.rrrr.field);
end.