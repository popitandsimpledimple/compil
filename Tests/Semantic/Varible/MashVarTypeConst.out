program
   main
   decls
      vars
         var
            e
            w
            type
               double
            no expression
         var
            x
            u
            type
               integer
            no expression
         var
            y
            m
            type
               string
            no expression
      consts
         const
            a
               integer
            5
         const
            b
            type
               string
            'hello world'
      decls types
         decl type
            int
            type
               integer
         decl type
            mystring
            type
               string
      vars
         var
            pp
               array
                  range
                     ..
                        5
                        20
            type
               integer
            no expression
         var
            oo
               array
                  range
                     ..
                        7
                        30
               array
                  range
                     ..
                        0
                        40
            type
               integer
            no expression
         var
            ii
               array
                  range
                     ..
                        5
                        50
               array
                  range
                     ..
                        0
                        60
               array
                  range
                     ..
                        0
                        70
            type
               integer
            no expression
      vars
         var
            rec
               record
               params
                     aaa
                     type
                        integer
                     dddd
                     type
                        double
                     bbbb
                     type
                        boolean
                     ssss
                     type
                        string
                     rrrr
                        record
                        params
                              field
                              type
                                 integer
            no expression
   block
      compounds statements
         call
            write
               array access
                  pp
                  4
               array access
                  array access
                     oo
                     10
                  10
               array access
                  array access
                     array access
                        ii
                        +
                           10
                           10
                     *
                        20
                        2
                  +
                     12
                     *
                        3
                        2
         call
            write
               record access
                  rec
                  dddd
               record access
                  rec
                  bbbb
               record access
                  rec
                  ssss
         call
            read
               record access
                  record access
                     rec
                     rrrr
                  field

 ───────────────────────────────────────────────────────────────────────────────────────────────────
 │ integer                   │ type                      │ integer                                 │
 │ char                      │ type                      │ char                                    │
 │ string                    │ type                      │ string                                  │
 │ double                    │ type                      │ double                                  │
 │ boolean                   │ type                      │ boolean                                 │
 ───────────────────────────────────────────────────────────────────────────────────────────────────

 ───────────────────────────────────────────────────────────────────────────────────────────────────
 │ main                      │ program name              │ no return type                          │
 │ e                         │ var                       │ double                                  │
 │ w                         │ var                       │ double                                  │
 │ x                         │ var                       │ integer                                 │
 │ u                         │ var                       │ integer                                 │
 │ y                         │ var                       │ string                                  │
 │ m                         │ var                       │ string                                  │
 │ a                         │ const                     │ integer                                 │
 │ b                         │ const                     │ string                                  │
 │ int                       │ alias                     │ integer                                 │
 │ mystring                  │ alias                     │ string                                  │
 │ pp                        │ var                       │ array                                   │
 │ oo                        │ var                       │ array                                   │
 │ ii                        │ var                       │ array                                   │
 │ rec                       │ var                       │ record                                  │
 ───────────────────────────────────────────────────────────────────────────────────────────────────
